src/common/telemetry/exception-telemetry-listener.ts

```diff
@@ -1,13 +1,11 @@
 // Copyright (c) Microsoft Corporation. All rights reserved.
 // Licensed under the MIT License.
-import { BaseActionPayload } from 'background/actions/action-payloads';
-import { TelemetryEventHandler } from 'background/telemetry/telemetry-event-handler';
 import { Logger } from 'common/logging/logger';
 import { escapeRegExp } from 'lodash';
 import * as TelemetryEvents from '../../common/extension-telemetry-events';
 import { UnhandledErrorTelemetryData } from '../../common/extension-telemetry-events';
 
-export class ExceptionTelemetryListener {
+export abstract class ExceptionTelemetryListener {
```

Comment:
In general, it's usually better to prefer composition over inheritance for a situation like there where two instances of a component operate the same way modulo one inner call's behavior. Using `protected abstract` inheritance like this works, but:

* It's substantially more overhead to write and to test. Consider how this PR would look if `ExceptionTelemetryListener` instead accepted a constructor parameter like `private readonly publishTelemetry: (eventName: string, data: UnhandledErrorTelemetryData) => void`. The entire `ForwardingExceptionTelemetryListener` class and its tests could be eliminated, the `SendingExceptionTelemetryListener` would be cut in half, and its tests would become substantially simpler (no longer needing to jump through a test subclass)
* The overhead gets even worse if you ever need to add a second dependency like this; if that happens, the inheritance-based approach needs `2^numDeps` concrete classes, where the extra code required for the composition-based approach scales linearly

The book [Design Patterns](https://www.amazon.com/Design-Patterns-Elements-Reusable-Object-Oriented/dp/9332555400#customerReviews) is an oldie-but-goodie for a deeper exploration of the tradeoffs (you can probably get a digital copy from the MS Library, there might also be a copy in the team room)